### PR TITLE
Move support for defining package managers to image definition

### LINKED ISSUE #400: package manager should possibly be selected via image.yaml / module yaml
Thanks to #328 we have `--package-manager`, but since the correct package manager is a property of the image being built, rather than the builder environment (running cekit), it should probably be something that can be put in a module or `image.yaml` property.

### PR DESCRIPTION
**Pull request by goldmann**
This removes support for defining package manager used to install packages
from the command line (and configuration file) to the image descriptor
itself.

Adds support for dnf.

Fixes #400

### CODE REVIEW

**Comment by goldmann** [comment]
I wonder if it would not make more sense to not specify the default at all and require the `manager` key when defining packages to install.

**Comment by goldmann** [comment]
On the other hand, having `yum` as the default value work in all Fedora/RHEL cases...

**Comment by luck3y** [comment]
> On the other hand, having `yum` as the default value works in all Fedora/RHEL cases...

+1, as its also the current default, it will just continue to work as expected.

File: Pipfile [hunk 1]
```diff
@@ -13,6 +13,8 @@ pytest-mock = "*"
 docker = "*"
 docker-squash = "*"
 click = "*"
+pylint = "<2.0.0"
```
Comment (rnc) [inline Pipfile]: Why < version 2 ? It seems F29 has 2.1.1 ...

**Comment by rnc** [comment]
Should this have documentation updates as well?

**Comment by goldmann** [comment]
Yes, I'll update this PR.

File: Pipfile (hunk 1 above)
Comment (goldmann) [reply Pipfile]: I missed it, it was automatically added by IDE.

**Comment by goldmann** [comment]
@rnc Docs added. I additionally worked to make it build without any warnings.

File: docs/descriptor/packages.rst [hunk 2]
```diff
@@ -16,12 +17,30 @@ section where you specify package names and repositories to be used.
...
 
-.. _repo:
+Package manager
+^^^^^^^^^^^^^^^^^^
+
+It is possible to define package manager used in the image
+used to install packages as part of the build process.
+
+.. note::
+    If you do not specify this key the default value is ``yum``.
```
Comment (rnc) [inline docs/descriptor/packages.rst]: Should this list possible options? e.g.
```suggestion
    If you do not specify this key the default value is ``yum``. Options are ``yum``, ``dnf``, and ``microdnf``
```

**Pull request merged**